.github/workflows/build_and_test.yaml

```diff
@@ -22,6 +22,8 @@ jobs:
       # lint
       - run: make lint-deps
       - run: make -k lint
+      # gen-check
+      - run: make -k gen-check
```

Comment:
Can this go in its own job, instead of in the `lint` job?

Reply:
I'd vote for option 3. adding it as a dependency to `lint`. As a dev I dont want to learn more make targets, would be ideal if `make lint` took care of this for me

Reply:
@LukeShu @arkodg do your guys make a conclusion？

Reply:
Please ignore my comment and consider Luke's

Reply:
done